packages/@aws-cdk/aws-servicecatalogappregistry/lib/aspects/stack-associator.ts

```diff
@@ -89,33 +107,43 @@ abstract class StackAssociatorBase implements IAspect {
     }
 
     if (node.account != this.application.env.account && !this.sharedAccounts.has(node.account)) {
-      this.application.shareApplication({
-        accounts: [node.account],
-        sharePermission: SharePermission.ALLOW_ACCESS,
-      });
+      if (this.enableCrossAccount) {
+        this.application.shareApplication({
+          accounts: [node.account],
+          sharePermission: SharePermission.ALLOW_ACCESS,
+        });
 
-      this.sharedAccounts.add(node.account);
+        this.sharedAccounts.add(node.account);
+      } else {
+        this.warning(node, 'Cross-account stack detected but application sharing and association will be skipped because cross-account option is not enabled.');
+        return;
+      }
     }
   }
 }
 
 export class CheckedStageStackAssociator extends StackAssociatorBase {
   protected readonly application: IApplication;
   protected readonly applicationAssociator?: ApplicationAssociator;
+  protected readonly enableCrossAccount: boolean;
 
-  constructor(app: ApplicationAssociator) {
+  constructor(app: ApplicationAssociator, props?: StackAssociatorBaseProps) {
     super();
     this.application = app.appRegistryApplication();
     this.applicationAssociator = app;
+    this.enableCrossAccount = props?.enableCrossAccount ?? false;
   }
+
 }
 
 export class StageStackAssociator extends StackAssociatorBase {
   protected readonly application: IApplication;
   protected readonly applicationAssociator?: ApplicationAssociator;
+  protected readonly enableCrossAccount: boolean;
```

Comment:
Should we add this to `StageStackAssociatorBase`, since we're using it on two separate subclasses, or are there more subclasses of `StackStackAssociatorBase` than these two?